sencha-workspace/packages/slate-cbl/src/view/student/DashboardController.js

```diff
@@ -0,0 +1,137 @@
+    ],
+
+    config: {
+        id: 'slate-cbl-teacher-dashboard', // workaround for http://www.sencha.com/forum/showthread.php?290043-5.0.1-destroying-a-view-with-ViewController-attached-disables-listen-..-handlers
+        control: {
+            '#': {
+//                contentareachange: 'refresh',
+//                progressrowclick: 'onProgressRowClick',
+                democellclick: 'onDemoCellClick',
+                render: 'onComponentRender'
+            }
+        }
+    },
+    
+    // lifecycle overrides
+    init: function() {
+        var me = this;
+
+        Ext.create('Ext.data.Store', {
+            storeId: 'cbl-competencies-loaded',
+            model: 'Slate.cbl.model.Competency'
+        });
+    },
+
+
+    //event handlers
+    onComponentRender: function(dashboardView) {
+        var me = this,
+            siteEnv = window.SiteEnvironment || {},
+            student = siteEnv.cblStudent && Ext.create('Slate.cbl.model.Student', siteEnv.cblStudent)
+            studentDashboardCompetenciesList = dashboardView.el.down('#studentDashboardCompetenciesList'),
+            contentArea = siteEnv.cblContentArea && Ext.create('Slate.cbl.model.ContentArea', siteEnv.cblContentArea),
+            competenciesStore = Ext.getStore('cbl-competencies-loaded'),
+            competenciesTpl = Ext.XTemplate.getTpl(me.view, 'competenciesTpl');
+
+        if (!student || !contentArea) {
+            return;
+        }
+
+        // empty competencies list
+        studentDashboardCompetenciesList.empty();
+        studentDashboardCompetenciesList.removeCls('competencies-unloaded').addCls('competencies-loading');
+
+        contentArea.getCompetenciesForStudents([student.getId()], function(competencies) {
+            var competenciesLength = competencies.length,
+                competencyIndex = 0,
+                competency, skillsList;
+            
+
+            if(!competenciesStore.isLoaded()) {
+                competenciesStore.loadRawData(competencies);
+            }
+
+            competenciesTpl.overwrite(studentDashboardCompetenciesList, {
+                student: student.getData(),
+                competencies: competencies
+            });
+
+            studentDashboardCompetenciesList.removeCls('competencies-loading').addCls('competencies-loaded');
+
+            for (; competencyIndex < competenciesLength; competencyIndex++) {
+                competency = Ext.create('Slate.cbl.model.Competency', competencies[competencyIndex]);
+                skillsList = studentDashboardCompetenciesList.down('.cbl-competency-panel[data-competency="'+competency.getId()+'"] .cbl-skill-meter');     
+                me.loadSkills(student, competency, skillsList);
+            }
+        });
+    },
+    
+    onDemoCellClick: function(dashboardView, ev, targetEl) {
+
+        Ext.create('Slate.cbl.view.student.skill.OverviewWindow', {
+            autoShow: true,
+            animateTarget: targetEl,
+
```

Comment:
Not stray, we're separating overrides to generic configs for the base class Ext.Window from the application-specific ones implemented in our OverviewWindow class